src/Datadog.Trace/Logging/DatadogLogging.cs

```diff
@@ -170,5 +143,49 @@ public static ILogger For<T>()
         {
             return GetLogger(typeof(T));
         }
+
+        private static string GetLogDirectory()
+        {
+            var nativeLogFile = Environment.GetEnvironmentVariable(ConfigurationKeys.ProfilerLogPath);
+            string logDirectory = null;
+
+            if (!string.IsNullOrEmpty(nativeLogFile))
+            {
+                logDirectory = Path.GetDirectoryName(nativeLogFile);
+            }
+
+            // This entire block may throw a SecurityException if not granted the System.Security.Permissions.FileIOPermission
+            // because of the following API calls
+            //   - Directory.Exists
+            //   - Environment.GetFolderPath
+            //   - Path.GetTempPath
+            if (logDirectory == null)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                    if (Directory.Exists(windowsDefaultDirectory))
+                    {
+                        logDirectory = windowsDefaultDirectory;
+                    }
+                }
+                else
+                {
+                    // either Linux or OS X
+                    if (Directory.Exists(NixDefaultDirectory))
```

Comment:
We should try to create the log folder if it doesn't already exist. We can do something like this (pseudo-code):
```
var logDirectory = GetFromSettings()

if logDirectory is null or empty
{
    logDirectory = isWindows ? windowsDefaultDirectory : linuxDefaultDirectory
}

if not exist logDirectory
{
    TryCreate(logDirectory)
}

if not exist logDirectory
{
    logDirectory = GetTempPath() + "/datadog-dotnet-tracer"
}
```

Reply:
We should also consider dropping the fallback to the temp path entirely. Let's discuss on Monday.

Reply:
I wasn't 100% sure that I could create the directory and apply permissions in Linux from within the application that the tracer is monitoring. I suppose it wouldn't hurt to try it, but I believe having a script that the user can run is more likely to be the most-used scenario since Linux users usually do not run with elevated permissions.

Reply:
I'm thinking of two cases:
- on Linux, if users override the default path to something the app will be able to write to, like `~/datadog-logs` or `/tmp/datadog-logs`
- on Windows, the default path (`ProgramData`) is usually writable

Don't worry about setting permissions: if we can create the directory, we can create files inside of it.

Reply:
You're making a case for dropping the temp directory logging fallback, right? That's fine. And this PR is only addressing Linux, not Windows. I want to keep them separate rather than muddying up the PR.